### ISSUE TITLE
Build the Software, Chapter 2

### ISSUE THREAD

**Issue by sulqadari** [ve3wwg/stm32f103c8t6#31]
Dear Mr. Gay,
kindly ask you to provide me with assistance to overcome step 'Build the Software', chapter 2.
I successfully went through preceding steps exactly the way you mentioned in your Book:
- cygwin installed;
- make and git installed and being recognized by cygwin as a commands;
- stm32 repo successfully cloned;
- libopencm3 cloned (NOTE: I checked out to commit dated oct 12, 2017, otherwise there is additional issue with unrecognized python dir location);
- FreeRTOS unzipped;
- variable FREERTOS assigned with the actual name of the related root folder;
- ARM cross compiller installed and being recognized by cygwin as a command (the lastest version);

But now I stuck at this last but one step (Build the Software).
the cygwin prints the below mentioned log:

codej@DESKTOP-5I7ASE8 /cygdrive/c/Lessons/stm32
$ make
make -C libopencm3 TARGETS=stm32/f1
make[1]: Entering directory '/cygdrive/c/Lessons/stm32/libopencm3'
  GENHDR  include/libopencm3/efm32/g/irq.json
  ...
  GENHDR  include/libopencm3/vf6xx/irq.json
  BUILD   lib/stm32/f1
/bin/sh: -c: line 0: unexpected EOF while looking for matching `"'
/bin/sh: -c: line 1: syntax error: unexpected end of file
make[1]: *** [Makefile:64: lib/stm32/f1] Error 1
make[1]: Leaving directory '/cygdrive/c/Lessons/stm32/libopencm3'
make: *** [Makefile:38: libopencm3/lib/libopencm3_stm32f1.a] Error 2

Kindly ask you to provide me with assistance.

**Comment by jegfish** [comment]
After 5 years perhaps it is too late to help you, but for anyone else who ends up here from a search engine, here is the fix I found.

You can get more information about the issue by just trying to build libopencm3 by itself, with `make V=1` for more details.

```sh
$ cd libopencm3
$ make V=1
```

This shows us:

```
make --directory=lib/stm32/f0 SRCLIBDIR="/home/user/stm32book/libopencm3/lib\"
/bin/sh: -c: line 1: unexpected EOF while looking for matching `"'
```

Somewhere we are generating a backslash in the command right before a quote, so it is escaped and not treated as a closing quote.

This is in `libopencm3/Makefile`, on this line:

```
SRCLIBDIR:= $(subst $(space),\$(space),$(realpath lib))
```

Edit this line to replace the backslash with a forward slash.

**Comment by sulqadari** [comment]
Yeah, it's a little bit outdated topic:) I had addressed that issue later, when switched to Ubuntu. But thank you anyway
